### ISSUE TITLE
Minecraft MAP Elites block count characterization

### ISSUE THREAD

**Issue by schrum2** [schrum2/MM-NEAT#769]
MAP Elites with Minecraft works! Of course, there is still work to be done, and it will only be as good as our novelty characterization and quality scores.

In the package `edu.southwestern.tasks.evocraft.characterizations` make a new class `MinecraftMAPElitesBlockCountBinLabels`. You can model it on `MinecraftMAPElitesWidthHeightDepthBinLabels`. The point of bin labels is to define a bunch of bins that MAP-Elites can store elite genomes in. This new set of labels counts the number of non-AIR blocks in a shape, and categorizes shapes based on that. In other words, for each possible number of blocks, try to find the best solution to some problem.

This set of labels should be fairly simple. There is only one dimension, and the size of that one dimension is the number of possible blocks in the shape (multiply the x, y, and z ranges). The `properties` should just be a list containing `OccupiedCountFitness`

**Comment by alejmedinajr** [comment]
There is now a loop that goes through the total possible number of blocks +1, and makes bin labels based on the number of blocks. The new bin labels scheme is running and we will see how it goes.

**Comment by schrum2** [comment]
Is the main method with the test code committed? I'm not seeing it

**Comment by alejmedinajr** [comment]
The main method with the test code is now committed.

**Comment by schrum2** [comment]
I'm currently testing this with volume generation instead of snake generation. However, I think the class could use some more comments from @alejmedinajr in the meantime.

**Comment by alejmedinajr** [comment]
More comments have been added.

**Comment by schrum2** [comment]
Once @MuellMark 's ability to view the archive in the world (#771) is fully working and merged, I would like you to set up and run this experiment on several machines in the lab. Log into some unused machines and get help from @MuellMark to install the needed software so you can have multiple machines testing this out.

In the meantime, my test seems to be going well. I'll upload some results later.

**Comment by schrum2** [comment]
It fills all 1001 bins:
![image](https://user-images.githubusercontent.com/8397236/172442003-b1cab896-7325-45b3-bb1e-493a53def7af.png)
This demonstrates that it was getting good coverage (before I terminated the run):
![image](https://user-images.githubusercontent.com/8397236/172442164-d9a18977-88ea-4694-b8a0-35614877e17a.png)
However, this also reveals that the max number of bins was being incorrectly calculated, but I've already committed a fix to this.
The fitness used was the Diversity Score, and that was still growing:
![image](https://user-images.githubusercontent.com/8397236/172442378-70586f49-b7ce-4dce-b143-cc71b14fd52b.png)


Basically, this seems to work, and I'm going to close the issue, but @alejmedinajr should still run additional trials of this on several lab machines to experiment.

**Comment by alejmedinajr** [comment]
Overnight test with 3d shapes and random seed 1:
![test with random seed 1](https://user-images.githubusercontent.com/100097809/172654618-e9f2b906-95fe-40a9-be66-bcb18e5852a3.png)

I looked at the other tests (with the other two random seeds) and they also have the same result (11 bins filled)

**Comment by schrum2** [comment]
This issue can remain closed, but here are some additional notes. 

The stack track reported above is actually related to a problem being handled in this issue #776 .

As for showing results @alejmedinajr , it makes sense that all 11 bins got filled right away, but look at other plots produced as well, such as the archive occupancy, the QD score, and max fitness

**Comment by alejmedinajr** [comment]
Here are some of the other plots as well. 
![MAPElitesBlockCountVolume_QDScores](https://user-images.githubusercontent.com/100097809/172696508-f8882716-7759-4f4f-ac20-27fe8b20094e.png)

![MAPElitesBlockCountVolume_MaxIndividualScores](https://user-images.githubusercontent.com/100097809/172696511-8ce1bb07-ac04-47fb-ab83-622b4f0912e5.png)

![MAPElitesBlockCountVolume_ArchivePerformance](https://user-images.githubusercontent.com/100097809/172696512-e0657aaa-30d2-4fcb-b84b-5d657a92f417.png)

![MAPElitesBlockCountVolume_ArchiveFilledBins](https://user-images.githubusercontent.com/100097809/172696514-0bb3a9bc-e2c0-48bc-be57-6e61db4a5c16.png)
